plugins/pagerduty/src/components/TriggerButton/index.test.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { act, fireEvent, waitFor } from '@testing-library/react';
+import { renderInTestApp } from '@backstage/test-utils';
+import {
+  ApiRegistry,
+  alertApiRef,
+  createApiRef,
+  ApiProvider,
+  IdentityApi,
+  identityApiRef,
+} from '@backstage/core';
+import { pagerDutyApiRef } from '../../api';
+import { Entity } from '@backstage/catalog-model';
+import { EntityProvider } from '@backstage/plugin-catalog-react';
+import { TriggerButton } from './';
+
+describe('TriggerButton', () => {
+  const mockIdentityApi: Partial<IdentityApi> = {
+    getUserId: () => 'guest@example.com',
+  };
+
+  const mockTriggerAlarmFn = jest.fn();
+  const mockPagerDutyApi = {
+    triggerAlarm: mockTriggerAlarmFn,
+  };
+
+  const apis = ApiRegistry.from([
+    [
+      alertApiRef,
+      createApiRef({
+        id: 'core.alert',
+        description: 'Used to report alerts and forward them to the app',
+      }),
+    ],
+    [identityApiRef, mockIdentityApi],
+    [pagerDutyApiRef, mockPagerDutyApi],
+  ]);
+
+  it('renders the trigger button, opens and closes dialog', async () => {
+    const entity: Entity = {
+      apiVersion: 'backstage.io/v1alpha1',
+      kind: 'Component',
+      metadata: {
+        name: 'pagerduty-test',
+        annotations: {
+          'pagerduty.com/integration-key': 'abc123',
+        },
+      },
+    };
+
+    const { queryByRole, getByRole, getByTestId } = await renderInTestApp(
```

Comment:
total side note, but i find it super convenient to consistently import `screen` from `@testing-library/react` and then always do `screen.getByRole` etc - no need to care about the return value of the render. It's what Dodds himself recommends apparently :P

Reply:
Thanks, as I said above I don't use the base testing lib for a while, was just following what the tests close to this did, but will start doing this way for sure.

Reply:
Yeah it's perfectly fine this way and arguably what most people do!